lib/package-managers/npm-proxy.js

```diff
@@ -43,6 +43,7 @@ if (proxyUrl) {
     proxyAgent = new ProxyAgent({
         host: proxy.hostname,
         port: proxy.port,
+        auth: proxy.username ? proxy.username + (proxy.password ? ':' + proxy.password : '') : proxy.auth,
```

Comment:
Although it might not happen in the wild, the `username` can be empty and only a password is set. So maybe check for `global.URL ?` instead.